addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -438,6 +439,37 @@ options.registry.WebsiteFormEditor = FormEditor.extend({
         }
         this._addHiddenField(value, fieldName);
     },
+    /**
+     * Prompts the user to save changes before being redirected
+     * towards an action specified in value
+     */
+    onClickCreateButton: function (previewMode, value, params) {
+        Dialog.confirm(this, _t("Would you like to save before being " +
```

Comment:
When an option method of the editor opens a Dialog, it should wait for that Dialog to be closed before resolving thee Promise it returns ("The `onClickCreateButton` method should not finish before the user has finished interacting with that method"). Check other examples opening a Dialog if I am not clear enough.